pantheon-decoupled.php

```diff
@@ -546,6 +571,116 @@ function pantheon_decoupled_delete_success() {
 	$wp_preview_delete->delete_preview_site( $delete_id );
 }
 
+function pantheon_decoupled_regen_env_vars() {
+  if ( ! current_user_can( 'manage_options' ) ) {
+    wp_die( esc_html__( 'You do not have sufficient permissions to access this page.', 'wp-decoupled-preview' ) );
+  }
+  check_admin_referer( 'env-regen', 'nonce' );
+
+  // Get data for preview site
+  $id = isset( $_GET['id'] ) ? absint( sanitize_text_field( $_GET['id'] ) ) : NULL;
+  $preview_sites = get_option( 'preview_sites' );
+  $preview_site = isset( $preview_sites['preview'][ $id ] ) ? $preview_sites['preview'][ $id ] : NULL;
+  
+  $docs_url = 'https://docs.pantheon.io/guides/decoupled/overview'
+    
+  ?>
+    <style>
+      #TB_window {
+        background-color: rgb(240, 240, 241);
+      }
+      #TB_window #adminmenumain,
+      #TB_window #wpfooter {
+        display: none;
+      }
+      #TB_window #wpcontent {
+        margin-left: 0;
+      }
+    </style>
+    <div class="wrap">
+        <h1><?php esc_html_e( 'Regenerate Environment Variables', 'wp-pantheon-decoupled' ); ?></h1>
+        <?php
+      if ( $id ) {
+        $site_label = $preview_site['label'];
+        $url = wp_nonce_url( add_query_arg( [
+          'page' => 'env_regen_action',
+          'id' => $id,
+        ], admin_url( 'options-general.php' ) ), 'env-regen', 'nonce' );
+
+        if(!$preview_site['associated_user']) {
+          echo '<p>A preview site must have an associated user to generate an application password.';
+        }
+        ?>
+
+        <a id="regen-password" <?= $preview_site['associated_user'] ? 'class="button-secondary button-large thickbox"' . 'href="' . esc_url( $url ) . '"' : 'class="button-secondary button-large"' .'disabled="true"' ?>>
```

Comment:
This isn't THAT hectic but lmk if there is a cleaner way to do this. The `thickbox` class needs to be optional here because if the button is disabled and is clicked with that class present it causes an infinite loading state of a modal that never gets rendered.